### DISCUSSION TITLE
7590AX : Image für aktuelle Inhouse-Version kann nicht gebaut werden wegen Kernel

### DISCUSSION THREAD

**Discussion by Sinus4** [Freetz-NG/freetz-ng#785] [Allgemein] [answered]
Beim Versuch, für die 7590AX ein Freetz-NG Image mit der aktuellen Inhouse-Version zu erstellen (die Labor-Version wird bei "menuconfig" nicht angeboten) erhalte ich folgende Fehlermeldung: 

`processing kernel
  checking kernel version
    ... used by AVM ...... 4.9.325
    ... used by Freetz ... 4.9.218
ERROR: Freetz and AVM kernel versions do not match. Wrong kernel version configuration in Freetz?`

Leider reichen meine Kenntnisse nicht aus um festzustellen, ob der Kernel 4.9.325 in den Quellen unter
[http://osp.avm.de/fritzbox/fritzbox-7590-ax/source-files-FRITZ.Box_7590_AX-grx5-07.50.tar.gz](http://osp.avm.de/fritzbox/fritzbox-7590-ax/source-files-FRITZ.Box_7590_AX-grx5-07.50.tar.gz)  enthalten sind.

**Comment by SaMMy-lacht** [comment, marked as answer]
Das os 7.50 und os 7.51 haben unterschiedliche Kernel, daher bringt Sourcecode nichts. Man bekommt auch keine labor oder inhaus Sourcecode von AVM (Ausnahmen sind die Kabelboxen).

Vielleicht baut es ja jemand ein. 
config/.img/Config.in
```
if FREETZ_TYPE_7599 && FREETZ_AVM_VERSION_07_5X
config FREETZ_AVM_HAS_WIREGUARD
	def_bool y
config FREETZ_AVM_HAS_AVMCOUNTERD
	def_bool y
config FREETZ_AVM_HAS_RRDTOOL
	def_bool y
config FREETZ_AVM_PROP_KERNEL_4_9_218
	def_bool y
endif
```
zu
```
if FREETZ_TYPE_7599 && FREETZ_AVM_VERSION_07_5X
config FREETZ_AVM_HAS_WIREGUARD
	def_bool y
config FREETZ_AVM_HAS_AVMCOUNTERD
	def_bool y
config FREETZ_AVM_HAS_RRDTOOL
	def_bool y
config FREETZ_AVM_PROP_KERNEL_4_9_325
	def_bool y
config FREETZ_AVM_PROP_KERNEL_CONFIG_AREA_SIZE_256
	def_bool y
endif
```
config/.img/separat//7599--07_5X.in
```
if FREETZ_TYPE_7599 && FREETZ_AVM_VERSION_07_5X

config FREETZ_AVM_HAS_WIREGUARD
	def_bool y

config FREETZ_AVM_HAS_AVMCOUNTERD
	def_bool y

config FREETZ_AVM_HAS_RRDTOOL
	def_bool y

config FREETZ_AVM_PROP_KERNEL_4_9_218
	def_bool y

endif


```
zu
```
if FREETZ_TYPE_7599 && FREETZ_AVM_VERSION_07_5X

config FREETZ_AVM_HAS_WIREGUARD
	def_bool y

config FREETZ_AVM_HAS_AVMCOUNTERD
	def_bool y

config FREETZ_AVM_HAS_RRDTOOL
	def_bool y

config FREETZ_AVM_PROP_KERNEL_4_9_325
	def_bool y

config FREETZ_AVM_PROP_KERNEL_CONFIG_AREA_SIZE_256
	def_bool y

endif
```
Danach kannst du aber erst mal keine 7.50 mehr für die 7590 AX erstellen. Da die ja wie gesagt ein anderen Kernel haben. 

Bei der 7590 und 7590 AX ist das zurzeit leider so, da beide halt auf 4_9_325 greifen in der inhaus und labor Version

**Comment by Sinus4** [comment]
Danke sehr für deine sehr gute Erläuterung, und dass du gleich eine Lösung mitgeliefert hast. Jetzt bin ich schlauer!

**Comment by fda77** [comment]
In Freetz haben all fws einer Version wie 07_5X nur 1 config datei. Wenn sich ein Feature wie die kernelversion ändert geht das so nicht wie es implementiert ist. Allerdings kann man so (für andere Geräte) inhaus bauen obwohl nur eine final eingebaut ist.
Ursache des Problems ist dass avm von inhaus>labor>final abgewichen ist, und es gar keine "stable" mehr gibt (waren eh immer buggy)
Am besten wäre es in freetz eine neu fritzos-version anzulegen- dann müsste man aber alle inhaus die man auswählen kann manuell hinzufügen. Ausserdem hat die die gleiche Version wie die letzte final.
Geht halt nicht

**Comment by Herbie-2005** [comment]
Vielen Dank für die Anleitung, SaMMy-lacht. Damit konnte ich ein neues Freetz-Image mit 154.07.51 rev104921 BETA bauen.

Für Openvpn bräuchte ich eine passende Datei tun.ko. Ich glaube, da steckt jetzt noch eine drin, die zum Kernel 4.9.218 paßt.
Wie komme ich am einfachsten zu einer tun.ko, die zum Kernel 4.9.325 paßt?

Gruß von Herbie

**Reply by SaMMy-lacht** [reply]
Ohne Sourcecode geht diese natürlich nicht

**Reply by Herbie-2005** [reply]
Ok, klar. Kann man da die Vorgängerversion von https://boxmatrix.info/wiki/Property:tun.ko nehmen und für die neue Umgebung neu kompilieren?
